packages/react/src/components/Tabs/tabs.test.tsx

```diff
@@ -0,0 +1,402 @@
+import React, { useRef } from 'react';
+import { render, screen, waitFor, fireEvent } from '@testing-library/react';
+import Tabs, { Tab, TabPanel } from './';
+import axe from '../../axe';
+
+const ariaLabel = 'I am a label';
+
+test('should render tabs without errors when mounted', async () => {
+  render(
+    <Tabs aria-label={ariaLabel}>
+      <div />
+    </Tabs>
+  );
+
+  expect(screen.getByRole('tablist').parentElement).toBeInTheDocument();
+});
+
+test('should render children with a tab', () => {
+  const TabswithRef = () => {
+    const ref = useRef(null);
+    return (
+      <>
+        <Tabs aria-label={ariaLabel}>
+          <Tab target={ref}>option 1</Tab>
+        </Tabs>
+        <TabPanel ref={ref}>
+          <p>Panel 1</p>
+        </TabPanel>
+      </>
+    );
+  };
+
+  render(<TabswithRef />);
+  expect(screen.getAllByRole('tab')).toHaveLength(1);
+});
+
+test('should only render tab components', () => {
+  render(
+    <Tabs aria-label={ariaLabel}>
+      <li>option 1</li>
+      <button>option 2</button>
+      <div className="no-show-div">option 3</div>
+    </Tabs>
+  );
+
+  expect(screen.queryByRole('listitem')).toBeNull();
+  expect(screen.queryByRole('button')).toBeNull();
+  expect(screen.queryByRole('generic', { name: /option 3/i })).toBeNull();
+});
+
+test('should render with thin prop', async () => {
+  const TabswithRef = () => {
+    const ref = useRef(null);
+    return (
+      <>
+        <Tabs aria-label={ariaLabel} thin>
+          <Tab target={ref}>option 1</Tab>
+        </Tabs>
+        <TabPanel ref={ref}>
+          <p>Panel 1</p>
+        </TabPanel>
+      </>
+    );
+  };
+
+  render(<TabswithRef />);
+  await waitFor(() => {
+    expect(screen.getByRole('tablist').parentElement).toHaveClass('Tabs--thin');
+  });
+});
+
+test('should pass classNames through', async () => {
+  const TabswithRef = () => {
+    const ref = useRef(null);
+    return (
+      <>
+        <Tabs aria-label={ariaLabel} className="find--me">
+          <Tab target={ref}>option 1</Tab>
+        </Tabs>
+        <TabPanel ref={ref}>
+          <p>Panel 1</p>
+        </TabPanel>
+      </>
+    );
+  };
+
+  render(<TabswithRef />);
+  await waitFor(() => {
+    expect(screen.getByRole('tablist').parentElement).toHaveClass('find--me');
+  });
+});
+
+test('should render with aria-label prop', () => {
+  const TabswithRef = () => {
+    const ref = useRef(null);
+    return (
+      <>
+        <Tabs aria-label="find-me">
+          <Tab target={ref}>option 1</Tab>
+        </Tabs>
+        <TabPanel ref={ref}>
+          <p>Panel 1</p>
+        </TabPanel>
+      </>
+    );
+  };
+
+  render(<TabswithRef />);
+  expect(screen.getByRole('tablist')).toHaveAttribute('aria-label', 'find-me');
```

Comment:
In this case, checking for the accessible name would be better since that's what the user expects:

```suggestion
  expect(screen.getByRole('tablist')).toHaveAccessibleName('aria-label');
```